lib/pages/scout/bloc/ScoutPageBloc.dart

```diff
@@ -1,37 +1,55 @@
 import 'package:ax_dapp/pages/scout/models/AthleteScoutModel.dart';
 import 'package:ax_dapp/pages/scout/models/ScoutPageState.dart';
 import 'package:ax_dapp/pages/scout/usecases/GetScoutAthletesDataUseCase.dart';
+import 'package:ax_dapp/service/Controller/usecases/GetChainChangesUseCase.dart';
 import 'package:ax_dapp/util/BlocStatus.dart';
 import 'package:ax_dapp/util/SupportedSports.dart';
+import 'package:ax_dapp/util/CurrentChain.dart';
 import 'package:flutter_bloc/flutter_bloc.dart';
 
 import '../models/ScoutPageEvent.dart';
 
 class ScoutPageBloc extends Bloc<ScoutPageEvent, ScoutPageState> {
   final GetScoutAthletesDataUseCase repo;
+  final GetChainChangesUseCase chainChangeUseCase;
+  late Stream _chainChangeStream;
 
-  ScoutPageBloc({required this.repo}) : super(ScoutPageState.initial()) {
+  ScoutPageBloc({required this.repo, required this.chainChangeUseCase})
+      : super(ScoutPageState.initial()) {
     on<OnPageRefresh>(_mapRefreshEventToState);
     on<SelectSport>(_mapSelectSportToState);
     on<OnAthleteSearch>(_mapSearchAthleteEventToState);
+    _chainChangeStream = chainChangeUseCase.stream;
   }
 
   void _mapRefreshEventToState(
       OnPageRefresh event, Emitter<ScoutPageState> emit) async {
     try {
       emit(state.copyWith(status: BlocStatus.loading));
-
+      _chainChangeStream.listen((event) {

```

Comment:
This listening event should be done on initialization of the block. Doing it this way means you are setting a new listener reference everytime you have a refresh event

Instead set up your bloc to subscribe to the stream events this way

Reply:
Listener is now moved into bloc initialization